### DISCUSSION TITLE
Supported Algorithms

### DISCUSSION THREAD

**Discussion by daniel-mezo** [webauthn4j/webauthn4j#613] [Q&A] [answered]
Hi @ynojima,

What is the heuristic by which certain algorithms are supported by the lib, but others not yet?
Currently I see these to be supported:
```
RS1 = new COSEAlgorithmIdentifier(-65535);
RS256 = new COSEAlgorithmIdentifier(-257);
RS384 = new COSEAlgorithmIdentifier(-258);
RS512 = new COSEAlgorithmIdentifier(-259);
ES256 = new COSEAlgorithmIdentifier(-7);
ES384 = new COSEAlgorithmIdentifier(-35);
ES512 = new COSEAlgorithmIdentifier(-36);
```
(source: https://github.com/webauthn4j/webauthn4j/blob/master/webauthn4j-core/src/main/java/com/webauthn4j/data/attestation/statement/COSEAlgorithmIdentifier.java)

However, for example the demo page of WebAuthn (webauthn.io) supports the followings (this data is returned to the JS client upon registration):
```
pubKeyCredParams: Array(10)
0: {type: 'public-key', alg: -7}
1: {type: 'public-key', alg: -35}
2: {type: 'public-key', alg: -36}
3: {type: 'public-key', alg: -257}
4: {type: 'public-key', alg: -258}
5: {type: 'public-key', alg: -259}
6: {type: 'public-key', alg: -37}
7: {type: 'public-key', alg: -38}
8: {type: 'public-key', alg: -39}
9: {type: 'public-key', alg: -8}
```

**Question:** Now, we don't really know why WebAuthn4j or webauthn.io choose to support an alg or not, so we were wondering whether additional algorithms are planned to be supported in the lib and why these particular ones were choosen in the first place?

**Bonus Question:** Since we are intending to configure the algs by their name instead of their value, maybe it could be up to your consideration adding a public static (or regular) constructor for the  `COSEAlgorithmIdentifier` class (e.g.: `create(String algorithmName)`) to make it possible to use either values or names? (Yes, we can and did implement our own mapper, but it would be cleaner this way. Please consider this merely a suggestion.)

**Comment by ynojima** [comment, marked as answer]
As of today, WebAuthn4J only supports algorithms which can be implemented with OpenJDK8 built-in JCA providers. PS256/384/PS512 are not supported by OpenJDK8. As far as I know, most authenticators in the real world use ES256 (only Windows Hello uses RS1), and there are no needs for aditional algorithms.
